lib/provider.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState, useContext, FC, PropsWithChildren } from 'react';
+import { App } from './types';
+
+const AppContext = React.createContext<App | null>(null);
```

Comment:
Could we also rename this to something more specific to the dashboard? Again, `AppContext` is too general